crates/fiber/src/windows.rs

```diff
@@ -0,0 +1,126 @@
+use crate::RunResult;
+use std::cell::Cell;
+use std::io;
+use std::ptr;
+use winapi::shared::minwindef::*;
+use winapi::um::fibersapi::*;
+use winapi::um::winbase::*;
+
+pub struct Fiber {
+    fiber: LPVOID,
+    state: Box<StartState>,
+}
+
+pub struct Suspend {
+    state: *const StartState,
+}
+
+struct StartState {
+    parent: Cell<LPVOID>,
+    initial_closure: Cell<*mut u8>,
+    result_location: Cell<*const u8>,
+}
+
+extern "C" {
+    fn wasmtime_fiber_get_current() -> LPVOID;
+}
+
+unsafe extern "system" fn fiber_start<F, A, B, C>(data: LPVOID)
+where
+    F: FnOnce(A, &super::Suspend<A, B, C>) -> C,
+{
+    let state = data.cast::<StartState>();
+    let func = Box::from_raw((*state).initial_closure.get().cast::<F>());
+    (*state).initial_closure.set(ptr::null_mut());
+    let suspend = Suspend { state };
+    let initial = suspend.take_resume::<A, B, C>();
+    super::Suspend::<A, B, C>::execute(suspend, initial, *func);
+}
+
+impl Fiber {
+    pub fn new<F, A, B, C>(stack_size: usize, func: F) -> io::Result<Fiber>
+    where
+        F: FnOnce(A, &super::Suspend<A, B, C>) -> C,
+    {
+        unsafe {
+            let state = Box::new(StartState {
+                initial_closure: Cell::new(Box::into_raw(Box::new(func)).cast()),
+                parent: Cell::new(ptr::null_mut()),
+                result_location: Cell::new(ptr::null()),
+            });
+            let fiber = CreateFiber(
```

Comment:
So I think `CreateFiberEx` should probably be used here as we probably want `stack_size` to be the reserve size and not the commit size on Windows.

`CreateFiber`'s `dwStackSize` parameter is the commit size and 1 MiB is typically used for the reserve size (it's from the PE header, but 1 MiB is what Microsoft's linker uses).

If `dwStackSize` is larger than the default reserve size, then it reserves `dwStackSize` rounded up 1 MiB as the reserve size.

With `CreateFiberEx`, there are two parameters: `dwStackCommitSize` and `dwStackReserveSize`. The former we'll want to pass 0 to get the default commit size (which will be 1 page in size, I believe).

So suggested change:

```rust
 let fiber = CreateFiberEx(
    0,
    stack_size,
    1, // FIBER_FLAG_FLOAT_SWITCH (might not be in the winapi crate)
    Some(fiber_start::<F, A, B, C>),
    &*state as *const StartState as *mut _,
);
```

This should help prevent the fiber stacks from counting towards the commit limit upfront.